### PR TITLE
chore(docs): readme file overhaul

### PR DESCRIPTION
**Pull request by peknur**
Clean up README file and move focus to official documentation which is
displayed on the Terraform Registry website (docs/index.md).

Move all developing instructions to DEVELOPING.md.

### CODE REVIEW

File: examples/provider/provider_env.tf [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  required_providers {
+    upcloud = {
+      source  = "UpCloudLtd/upcloud"
+      version = "~> 2.0"
+    }
+  }
+}
+
+provider "upcloud" {
+}
```
Comment (Darep) [inline examples/provider/provider_env.tf:11]: Is this example missing some env variables?

Comment (Darep) [reply examples/provider/provider_env.tf:11]: If it's meant to be like this, then maybe add some comment here to make it clear 🤔 

```suggestion
  # this is left empty intentionally, UpCloud credentials are read from the environment variables UPCLOUD_USERNAME and UPCLOUD_PASSWORD
}
```

File: README.md [hunk 2]
```diff
@@ -1,372 +1,21 @@
...
 
 # UpCloud Terraform Provider
-
-[![tests](https://github.com/UpCloudLtd/terraform-provider-upcloud/workflows/UpCloud%20Terraform%20provider%20tests/badge.svg)](https://github.com/UpCloudLtd/terraform-provider-upcloud/actions)
-
-* Website: https://www.terraform.io
-* [![Gitter chat](https://badges.gitter.im/hashicorp-terraform/Lobby.png)](https://gitter.im/hashicorp-terraform/Lobby)
-* Mailing list: [Google Groups](http://groups.google.com/group/terraform-tool)
-* Check [examples](./examples) directory
+[![tests](https://github.com/UpCloudLtd/terraform-provider-upcloud/actions/workflows/terraform.yml/badge.svg?event=pull_request&branch=)](https://github.com/UpCloudLtd/terraform-provider-upcloud/actions/workflows/terraform.yml)
```
Comment (Darep) [inline README.md:6]:
```suggestion
```

I'd suggest we remove the badge also completely for now :)

File: examples/provider/provider_env.tf (hunk 1 above)
Comment (peknur) [reply examples/provider/provider_env.tf:11]: Those examples are included to `docs/index.md` by tfplugindocs and those code snippets are documented there. But you are right those examples on their own won't make much sense :) I'll add more comments to those tf files also.

File: docs/index.md [hunk 3]
```diff
@@ -9,22 +9,44 @@ description: |-
...
+      version = "~> 2.0"
+    }
+  }
+}
+
+provider "upcloud" {
+  # username and password configuration arguments can be omitted  
+  # if environment variables UPCLOUD_USERNAME and UPCLOUD_PASSWORD are set
+  # username = ""
+  # password = ""
```
Comment (ka-myl) [inline docs/index.md:29]: I would also put this comment in the `minimal` example, as that is probably the first one users would check

**Review by ka-myl — APPROVED** [review]
LGTM

**Pull request merged**